https://bugs.gentoo.org/403593

the merging of code in upstream commit 2244d1a9877d18bdc42a5204d31f41920d369e87
introduced this bug.  patch by Nikita Kozlov.

--- a/config.c
+++ b/config.c
@@ -1045,7 +1045,7 @@ struct mddev_ident *conf_match(struct supertype *st,
 					array_list->devname);
 			continue;
 		}
-		if (array_list->devices &&
+		if (array_list->devices && devname &&
 		    !match_oneof(array_list->devices, devname)) {
 			if (verbose >= 2 && array_list->devname)
 				fprintf(stderr, Name
